### PR TITLE
Change to use native vscode clipboard

### PR DESCRIPTION
**Pull request by xconverge**

### CODE REVIEW

**Comment by xconverge** [comment]
fixes #1812

**Comment by jpoon** [comment]
Sweet. Can we also get rid of `overrideCopy`?

File: package.json [hunk 1]
```diff
@@ -674,7 +674,6 @@
   },
   "dependencies": {
     "appdirectory": "0.1.0",
-    "clipboardy": "1.2.3",
```
Comment (jpoon) [inline package.json]: Update https://github.com/VSCodeVim/Vim#linux-setup

File: src/util/clipboard.ts [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
-      clipboardy.writeSync(text);
+      await vscode.env.clipboard.writeText(text);
     } catch (e) {
       logger.error(e, `Clipboard: Error copying to clipboard. err=${e}`);
     }
   }
 
-  public static Paste(): string {
-    return clipboardy.readSync();
+  public static Paste(): Thenable<string> {
```
Comment (jpoon) [inline src/util/clipboard.ts]: return Promise instead?

Comment (xconverge) [reply src/util/clipboard.ts]: I actually am not really sure how to properly implement a thenable interface (which the clipboard read is) as a promise. Any help would be much appreciated...

Comment (jpoon) [reply src/util/clipboard.ts]: Should just be:

```suggestion
  public static async Paste(): Promise<string> {
    return vscode.env.clipboard.readText();
  }
```

**Comment by xconverge** [comment]
> 
> 
> Sweet. Can we also get rid of `overrideCopy`?
> 
> Edit: maybe not, we have a bunch of special logic there --
> 
> [Vim/src/actions/commands/actions.ts](https://github.com/VSCodeVim/Vim/blob/0151438b4b802f6123eac1864886aeba8de5e92b/src/actions/commands/actions.ts#L992)
> 
> Line 992 in [0151438](/VSCodeVim/Vim/commit/0151438b4b802f6123eac1864886aeba8de5e92b)
>  class CommandOverrideCopy extends BaseCommand {

Yea it that functionality is really for a different purpose unfortunately

File: src/util/clipboard.ts [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
+      await vscode.env.clipboard.writeText(text);
     } catch (e) {
       logger.error(e, `Clipboard: Error copying to clipboard. err=${e}`);
     }
   }
 
-  public static Paste(): string {
-    return clipboardy.readSync();
+  public static async Paste(): Promise<string> {
+    return vscode.env.clipboard.readText();
```
Comment (xconverge) [inline src/util/clipboard.ts]: do I want to await here?

**Comment by xconverge** [comment]
Tested on Win10 and Ubuntu

**Comment by jpoon** [comment]
Cutting a new release

**Pull request merged**